system/View/Traits/DecoratesViews.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace CodeIgniter\View\Traits;
+
+use CodeIgniter\View\Exceptions\ViewException;
+use CodeIgniter\View\Interfaces\ViewDecorator;
+
+use function config;
+
+trait DecoratesViews
+{
+    /**
+     * Runs the generated output through and declared
+     * view decorators.
+     *
+     * @param string $html
+     *
+     * @return string
+     */
+    private function decorateOutput(string $html): string
+    {
+        $decorators = config('View')->decorators;
+
+        if (empty($decorators)) {
+            return $html;
+        }
+
+        foreach ($decorators as $className) {
+            $decorator = new $className();
+
+            if (! $decorator instanceof ViewDecorator) {
+                throw ViewException::forInvalidDecorator($className);
+            }
+
+            $html = $decorator->decorate($html);
```

Comment:
This code can be optimized.
```suggestion
        foreach ($decorators as $decorator)
            if (! is_subclass_of($decorator, ViewDecorator::class) {
                throw ViewException::forInvalidDecorator($decorator);
            }
            
            $html = (new $decorator())->decorate($html);
```

Reply:
If we go with the static method suggestion above this can be simplified even more:
```suggestion
        foreach ($decorators as $className) {
            if (! is_subclass_of($decorator, ViewDecorator::class) {
                throw ViewException::forInvalidDecorator($className);
            }

            $html = $decorator::decorate($html);
```

Reply:
I like the static method suggestion here. Will change and simplify.